python/pyiceberg/cli/console.py

```diff
@@ -361,3 +361,24 @@ def table(ctx: Context, identifier: str, property_name: str):  # noqa: F811
         ctx.exit(1)
     else:
         raise NoSuchPropertyException(f"Property {property_name} does not exist on {identifier}")
+
+
+@run.group()
+def scan():
+    """Create a table scan."""
+
+
+@scan.command("table")
+@click.argument("identifier")
+@click.pass_context
+@catch_exception()
+def scan_table(ctx: Context, identifier: str):
```

Comment:
I like that we can easily visualize the files. I think we should merge this one with the files command since they are so similar. WDYT?

```
➜  python git:(master) ✗ pyiceberg --verbose true --catalog fokko files hive_test.h1
Snapshots: fokko.hive_test.h1
└── Snapshot 4362649644256929116, schema 0: 
    s3://bucket/41246f47-e1bc-465e-89b4-6d039a9d55dc/67ad77ac-d90b-4472-b1e9-672d91a1ac41/metadata/snap-4362649644256929116-1-63e88d4d-1b30-4cae-b7a0-f53b0b54758a.avro
    ├── Manifest: s3://bucket/41246f47-e1bc-465e-89b4-6d039a9d55dc/67ad77ac-d90b-4472-b1e9-672d91a1ac41/metadata/63e88d4d-1b30-4cae-b7a0-f53b0b54758a-m0.avro
    │   └── Datafile: 
    │       s3://bucket/41246f47-e1bc-465e-89b4-6d039a9d55dc/67ad77ac-d90b-4472-b1e9-672d91a1ac41/data/0bfca5ff//00000-0-dweeks_20220901095219_f181f97a-43ce-4aa6-8cec-6956d83b7a42-job_lo
    │       cal2002913683_0002-00001.parquet
    └── Manifest: s3://bucket/41246f47-e1bc-465e-89b4-6d039a9d55dc/67ad77ac-d90b-4472-b1e9-672d91a1ac41/metadata/508fd9ac-520f-47d8-900f-00f1dda051ac-m0.avro
        └── Datafile: s3://bucket/41246f47-e1bc-465e-89b4-6d039a9d55dc/67ad77ac-d90b-4472-b1e9-672d91a1ac41/data/3d030bcf/00000-0-44897130-c494-4dca-a164-e874c1e37c36-00001.parquet
```

Reply:
Not sure I follow you @Fokko . Do you mean make `scan` a sub-command of `files` command? Or just have `files` command use the `table.new_scan().plan_files()` API underneath?

If it's the later, I like that.

Reply:
I think the later one is more flexible, but then we should remove the files because they have so much overlap

Reply:
The only difference being `files` at present lists all snapshots. The table scan `plan_files()` will only list the snapshot that is being used for the scan.

Reply:
Maybe we can make files with three options:
- `files --snapshot=all` - The current files behavior
- `files --snapshot=current` - Files under the current snapshot
- `files --snapshot=<snapshot_id>` - Files under the snapshot_id specified